Code/Framework/AzCore/Tests/Math/QuaternionTests.cpp

```diff
@@ -408,4 +408,98 @@ namespace UnitTest
         Matrix4x4 m = Matrix4x4::CreateFromQuaternion(rotQuat);
         AZ_TEST_ASSERT(m.IsClose(rotMatrix));
     }
+
+    void TestQuaternionScaledAxisAngleRoundtrip(AZ::Quaternion testQuat)
+    {
+        // Take the shortest path for quaternions containing rotations bigger than 180.0°.
+        if (testQuat.GetW() < 0.0f)
+        {
+            testQuat = -testQuat;
+        }
+
+        // Scaled axis-angle
+        // Convert test quaternion to scaled axis-angle representation.
+        const AZ::Vector3 scaledAxisAngle = testQuat.ConvertToScaledAxisAngle();
+        {
+            // Convert the scaled axis-angle back into a quaternion.
+            AZ::Quaternion backFromScaledAxisAngle = AZ::Quaternion::CreateFromScaledAxisAngle(scaledAxisAngle);
+
+            // Compare the original quaternion with the one after the conversion to the scaled axis-angle representation.
+            EXPECT_TRUE(testQuat.IsClose(backFromScaledAxisAngle, 1e-6f));
+        }
+
+        // Axis-angle
+        {
+            AZ::Vector3 axis;
+            float angle;
+            testQuat.ConvertToAxisAngle(axis, angle);
+
+            AZ::Vector3 scaledResult = axis*angle;
+            EXPECT_TRUE(scaledResult.IsClose(scaledAxisAngle, 1e-5f));
+
+            // Special case handling for identity rotation.
+            AZ::Vector3 axisFromScaledResult = scaledResult.GetNormalized();
+            float angleFromScaledResult = scaledResult.GetLength();
+            if (AZ::IsClose(angleFromScaledResult, 0.0f))
+            {
+                axisFromScaledResult = AZ::Vector3::CreateAxisY();
+            }
+
+            const AZ::Quaternion backFromAxisAngle = AZ::Quaternion::CreateFromAxisAngle(axisFromScaledResult, angleFromScaledResult);
+            EXPECT_TRUE(testQuat.IsClose(backFromAxisAngle, 1e-6f));
+        }
+    }
+
+    void TestQuaternionScaledAxisAngleRoundtripRotationCombinations(float angle)
+    {
+        TestQuaternionScaledAxisAngleRoundtrip(AZ::Quaternion::CreateRotationX(angle));
```

Comment:
This is a lot of checks in one test again. If this ever fails it will be tricky to track down exactly where it's failing. I think it might also be a bit overkill in terms of the number of checks. Personally I think it would be sufficient to call the Roundtrip test on just a few carefully chosen quaternions.

Reply:
Changed into less, more carefully chosen edge cases for the parameterized tests.